jobsdb/jobsdb.go

```diff
@@ -2027,17 +2027,17 @@ func (jd *HandleT) internalStoreWithRetryEachInTx(tx *sql.Tx, ds dataSetT, jobLi
 	queryStat.Start()
 	defer queryStat.End()
 
-	_, err := tx.Exec(savepointSql)
+	_, err = tx.ExecContext(ctx, savepointSql)
 	if err != nil {
-		return failAll(err)
+		return failAll(err), err
```

Comment:
we only want to return a non nil err from this method if the error is `context.DeadlineExceeded`. All other errors shouldn't be considered to be recoverable